src/backend/stream.rs

```diff
@@ -414,6 +414,47 @@ impl<'ctx> PulseStream<'ctx> {
             }
         }
 
+        fn get_default_sink_name(
+            mainloop: &pulse::ThreadedMainloop,
+            context: &pulse::Context,
+        ) -> String {
+            struct CallbackData<'a> {
+                mainloop: &'a pulse::ThreadedMainloop,
+                default_sink_name: String,
+            }
+
+            fn server_info_cb(
+                _context: &pulse::Context,
+                info: Option<&pulse::ServerInfo>,
+                u: *mut c_void,
+            ) {
+                let r = unsafe { ptr::NonNull::new(u as *mut CallbackData).unwrap().as_mut() };
+
+                let name = unsafe { CStr::from_ptr(info.unwrap().default_sink_name) };
+                let name_slice = name.to_bytes_with_nul();
+                r.default_sink_name = std::str::from_utf8(name_slice).unwrap().to_string();
```

Comment:
```suggestion
                r.default_sink_name = name.to_str().unwrap().to_owned();
```
This should do the same thing (for utf8).

But I wonder what happens if `from_utf8(...)` / `to_str()` fails? Should a default name be assigned? or simply returning an error for `stream_init`?

Reply:
Yeah, that's a valid consideration. I'm thinking of making the default_sink_name an Result<String>, to indicate whether or not the pulse API was successfully able to find a default sink. If not, should we abort the function call (since technically "Loopback" failed)?

Reply:
Also I'd advise AGAINST using &str there as the CString's (by that I mean the ``char*`` type not the Rust one) state is unknown after the callback function is finished. Better be safe and just copy it into a String.